src/hooks/abilities/useAbilities.ts

```diff
@@ -0,0 +1,74 @@
+import { useMemo } from 'react'
+import { useTranslation } from 'react-i18next'
+
+import { checkETH2LDFromName } from '@app/utils/utils'
+
+import { useResolverIsAuthorized } from '../resolver/useResolverIsAuthorized'
+import { useAccountSafely } from '../useAccountSafely'
+import { useBasicName } from '../useBasicName'
+import { useHasSubnames } from '../useHasSubnames'
+import { getDeleteAbilities } from './utils/getDeleteAbilities'
+import { getEditAbilities } from './utils/getEditAbilities'
+import { getReclaimAbilities } from './utils/getReclaimAbilities'
+import { getSendAbilities } from './utils/getSendAbilities'
+
+export const useAbilities = (name: string) => {
+  const { t } = useTranslation('profile')
+  const parent = name?.split('.')?.slice(1).join('.')
+
+  const { address } = useAccountSafely()
+  const basicNameData = useBasicName(name, { skipGraph: false })
+  const resolverAuthorisation = useResolverIsAuthorized(name, {
+    enabled: !!name,
+  })
+  const parentBasicNameData = useBasicName(parent, { skipGraph: false, enabled: !!parent })
+
+  const hasSubnamesData = useHasSubnames(name)
```

Comment:
is this hook loaded at any point when the user isn't connected, if so these should all be conditionally enabled by it